src/contract.ts

```diff
@@ -0,0 +1,279 @@
+import BN from 'bn.js';
+import { Account, Connection } from 'near-api-js';
+import { FinalExecutionOutcome } from 'near-api-js/lib/providers';
+import {
+    CodeResult,
+    getTransactionLastResult,
+} from 'near-api-js/lib/providers/provider';
+import { ArgumentTypeError } from 'near-api-js/lib/utils/errors';
+import { ABI, ABIParameterInfo } from './abi';
+
+export interface FunctionCallOptions {
+    /** max amount of gas that method call can use */
+    gas?: BN;
+    /** amount of NEAR (in yoctoNEAR) to send together with the call */
+    attachedDeposit?: BN;
+    /**
+     * Metadata to send the NEAR Wallet if using it to sign transactions.
+     * @see {@link RequestSignTransactionsOptions}
+     */
+    walletMeta?: string;
+    /**
+     * Callback url to send the NEAR Wallet if using it to sign transactions.
+     * @see {@link RequestSignTransactionsOptions}
+     */
+    walletCallbackUrl?: string;
+}
+
+function deserializeJSON(response: Uint8Array): any {
+    return JSON.parse(Buffer.from(response).toString());
+}
+
+function serializeJSON(input: any): Buffer {
+    return Buffer.from(JSON.stringify(input));
+}
+
+// Helper function so that
+function ignoreSerialization(input: Buffer): Buffer {
+    return input;
+}
+
+/**
+ * Validation on arguments being a big number from bn.js
+ * Throws if an argument is not in BN format or otherwise invalid
+ * @param argMap
+ */
+function validateBNLike(argMap: { [name: string]: any }) {
+    const bnLike = 'number, decimal string or BN';
+    for (const argName of Object.keys(argMap)) {
+        const argValue = argMap[argName];
+        if (argValue && !BN.isBN(argValue) && isNaN(argValue)) {
```

Comment:
Oh, I didn't realize that `isNaN` is not literally something like `arg === NaN`. This check has some weird side-effects then like allowing an empty string (and even NaN itself funnily enough) to pass through. But maybe since JS considers empty strings as numbers so should `naj` :)

In any case yeah this discussion does not belong to this specific repo anyway